src/pollen.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Pollen charm business logic."""
+
+import glob
+import subprocess
+from pathlib import Path
+
+from charms.operator_libs_linux.v0 import apt
+from charms.operator_libs_linux.v1 import systemd
+from charms.operator_libs_linux.v2 import snap
+
+from exceptions import ConfigurationWriteError, InstallError
+
+SNAP_NAME = "gtrkiller-pollen"
+
+
+class PollenService:
+    """Pollen service class."""
+
+    def prepare(self, unit_name, charm_state) -> None:
+        """Install packages and write configuration files.
+
+        Args:
+            unit_name: ¨Pollen charm's unit name.
+            charm_state: Pollen charm's CharmState instance.
+
+        Raises:
+            InstallError: if the packages fail to install
+            ConfigurationWriteError: something went wrong writing the configuration
+        """
+        try:
+            apt.update()
+            apt.add_package(["pollinate", "ent"])
```

Comment:
The `ent` package is not needed anymore since the pollen with metrics using pure go to calculate metrics from the responses. And I don't think `pollinate` is needed either.

Reply:
Oh yes indeed